cosi/protocol/doc.go

```diff
@@ -0,0 +1,34 @@
+/*
+Package protocol contains an implementation of the CoSi protocol as described in the paper
+"Keeping Authorities "Honest or Bust" with Decentralized Witness Cosigning"
+
+The protocol has four messages:
+	- Announcement which is sent from the root down the tree and announce the proposal
+	- Commitment which is sent back up to the root, containing an aggregated commitment from all nodes
+	- Challenge which is sent from the root down the tree and contains the aggregated challenge
+	- Response which is sent back up to the root, containing the final aggregated signature, then used by the root to sign the proposal
+
+The protocol uses five files:
+- struct.go defines the messages sent around and the protocol constants
```

Comment:
for godoc to look nice, this has to be indented. We use two-space indents, please adjust also the lines above ("the protocol has four messages:...")